### DISCUSSION TITLE
Why c++?

### DISCUSSION THREAD

**Discussion by nxmxsxs** [CuarzoSoftware/Louvre#22] [General]
Hello, I'm just curious to know why this was implemented in c++. I assume it was due to the usage of classes/virtual methods as part of the standard API. Are there other reasons I might be missing? Thanks!

**Comment by ehopperdietzel** [comment]
Hello, there's no profound background behind my choice. I simply love C++. I also have a deep affection for C, so being able to use both is like a dream. Additionally, it allows you to include almost anything in the project. I've been learning more about Rust, and it would have been an excellent choice for this project, especially because it's very effective at preventing issues that could cause the compositor to crash. I don't rule out the option of porting it to Rust, although it would be a tedious task, I don't see it as impossible.
